ui/src/layout/signin/RootComponent.js

```diff
@@ -51,153 +46,144 @@ class TiDBSignInForm extends React.PureComponent {
         is_tidb_auth: true,
       })
       authUtil.setAuthToken(r.data.token)
-      message.success(this.props.t('signin.message.success'))
-      singleSpa.navigateToUrl('#' + this.props.registry.getDefaultRouter())
+      message.success(t('signin.message.success'))
+      singleSpa.navigateToUrl('#' + registry.getDefaultRouter())
     } catch (e) {
       console.log(e)
       if (!e.handled) {
         let msg
         if (e.response.data) {
-          msg = this.props.t(e.response.data.code)
+          msg = t(e.response.data.code)
         } else {
           msg = e.message
         }
-        this.setState({
-          signInError: this.props.t('signin.message.error', { msg }),
-        })
-        this.refForm.current.setFieldsValue({ password: '' })
+        setSignInError(t('signin.message.error', { msg }))
+        refForm.current.setFieldsValue({ password: '' })
         setTimeout(() => {
           // Focus after disable state is removed
-          this.refPassword.current.focus()
+          refPassword.current.focus()
         }, 0)
       }
     }
-    this.setState({ loading: false })
+    setLoading(false)
   }
 
-  handleSubmit = (values) => {
-    this.signIn(values)
+  const handleSubmit = (values) => {
+    signIn(values)
   }
 
-  clearErrorMessages = () => {
-    this.setState({ signInError: null })
+  const clearErrorMessages = () => {
+    setSignInError(null)
   }
 
-  componentDidMount = () => {
-    this.refPassword.current.focus()
-  }
+  useEffect(() => {
+    refPassword.current.focus()
+  })
```

Comment:
should be `useEffect(() => ..., [])` instead?

Reply:
Yep, @gauss1314 , can you fix this?

Reply:
Oh, by passing an empty array, only execute the side effect once (at mount time). Is that right? I will raise a PR to fix it.